### PR TITLE
m always sparse, dp and dm always hyper

### PR DESCRIPTION
**Pull request by swilly22**
It's not enough to specify the initial sparsity type, we need to prevent GraphBLAS from converting matrix format later on, this PR retains `M`, `DP` and `DM` format throughout their lifetime

### CODE REVIEW

**Review by AviAvni — APPROVED** [review]

File: src/graph/graph.c [hunk 1]
```diff
@@ -260,16 +260,78 @@ void Graph_SetMatrixPolicy
...
+	}
+
+	n = array_len(g->relations);
+	for(int i = 0; i < n; i ++) {
+		M = Graph_GetRelationMatrix(g, i, false);
+		RG_Matrix_wait(M, force_flush);
+	}
+}
+
+bool Graph_Pending
```
Comment (AviAvni) [inline src/graph/graph.c:287]: why not return immediately when any matrix is pending?

File: src/graph/rg_matrix/rg_pending.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+	GrB_Info    info;
+	bool        p        =  false;
+	bool        res      =  false;
+	GrB_Matrix  M        =  RG_MATRIX_M(C);
+	GrB_Matrix  DP       =  RG_MATRIX_DELTA_PLUS(C);
+	GrB_Matrix  DM       =  RG_MATRIX_DELTA_MINUS(C);
+
+	if(RG_MATRIX_MAINTAIN_TRANSPOSE(C)) {
+		info = RG_Matrix_pending(C->transposed, &res);
```
Comment (AviAvni) [inline src/graph/rg_matrix/rg_pending.c:26]: if transposed matrix has pending then no need to check non transposed

File: tests/unit/test_rg_matrix.cpp [hunk 3]
```diff
@@ -1428,3 +1450,79 @@ TEST_F(RGMatrixTest, RGMatrix_mxm) {
 	RG_Matrix_free(&D);
 	ASSERT_TRUE(C == NULL);
 }
+
+//#ifndef RG_DEBUG
```
Comment (AviAvni) [inline tests/unit/test_rg_matrix.cpp:1454]: why this commented

**Review by AviAvni — COMMENTED** [review]
looks good 3 small comments

File: src/graph/graph.c (hunk 1 above)
Comment (swilly22) [reply src/graph/graph.c:287]: 👍

File: src/graph/rg_matrix/rg_pending.c (hunk 2 above)
Comment (swilly22) [reply src/graph/rg_matrix/rg_pending.c:26]: 👍

**Pull request merged**